File: framework/languages/java/samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/play/actors/PlayActorFactory.kt
package systems.zlink.samples.kotlin.tictactoe.server.play.actors

import systems.zlink.framework.actors.ZLinkActor
import systems.zlink.framework.actors.ZLinkActorContext
import systems.zlink.framework.kotlin.ZLinkCoroutineActorFactory
import systems.zlink.framework.kotlin.ZLinkCoroutineRuntime

class PlayActorFactory(
    coroutines: ZLinkCoroutineRuntime,
) : ZLinkCoroutineActorFactory(coroutines) {
    override suspend fun create(
        actorId: String,
        context: ZLinkActorContext,
    ): ZLinkActor =
        PlayActor(actorId, context)
}

File: framework/languages/node/samples/TicTacToe.Ts/Server/Play/Adapters/ZLink/Actors/play-actor-factory.ts
import { PlayActor } from './play-actor';
import { actorDisplayName } from '../../../../../Shared/Contracts/messages';
import type {
  ZLinkActor,
  ZLinkActorContext
} from '@zlink-systems/framework';
import type {
  TicTacToeActor
} from '../../../../../Shared/Contracts/messages';

type PlayZLinkActor = TicTacToeActor & ZLinkActor;

class PlayActorFactory {
  create(actorId: string, context: ZLinkActorContext): PlayZLinkActor {
    return new PlayActor(actorId, actorDisplayName(actorId), context);
  }
}

export { PlayActorFactory };
